pkg/csi/cinder/openstack/openstack_volumes.go

```diff
@@ -396,6 +397,16 @@ func (os *OpenStack) GetMaxVolLimit() int64 {
 	return defaultMaxVolAttachLimit
 }
 
+// GetFreeCapacity returns free capacity of the block storage
+func (os *OpenStack) GetFreeCapacity() (int64, error) {
+	res, err := volumelimit.Get(os.blockstorage).Extract()
```

Comment:
I'm not 100% sure, but this sounds like an admin API. We can't make admin calls from CSIs, all CPO is supposed to work normally on a regular tenant.

Reply:
https://docs.openstack.org/api-ref/block-storage/v3/#limits-limits i did not find the role/permission list for csi-cinder.
I think this feature should be optional.

Reply:
Can you confirm it with DevStack and demo tenant?

Reply:
OVH Cloud, openstack account has `computeOperator`, `volumeOperator` permission, here doc: https://help.ovhcloud.com/csm/en-public-cloud-authenticate-api-openstack-service-account?id=kb_article_view&sysparm_article=KB0059364

but i do not know/can get the real openstack permission of it.
If you have a permission list of DevStack  - i'd like to check...

Reply:
I've checked that the API is available. What puzzles me now is - how often is `GetCapacity()` called? Would calling this API be a significant blow on the Cinder API in the cloud?

/ok-to-test

Reply:
it calls 1/min (`capacity-poll-interval` flag of csi-provisioner https://github.com/kubernetes-csi/external-provisioner/blob/c7f94435bd29e49edf1af0eda9c4ee2907c4d160/cmd/csi-provisioner/csi-provisioner.go#L111) for each storageClass and accessibleTopology (av zone).

flag `enable-capacity` by default is off, so csi-provisioner does not collect this metrics by default.